source/Outfit.h

```diff
@@ -105,6 +105,7 @@ class Outfit : public Weapon {
 	std::vector<std::string> licenses;
 	
 	Dictionary attributes;
+	std::map<std::string, std::map<double, int>> partialTerms;
```

Comment:
The code here looks quite elegant, but please add a few comments here on how the partialTerms help to avoid the rounding errors (because I expect that it is a bit hard to follow for people that just encounter this code later out of the context of this PR).